### ISSUE TITLE
Error duaring configurating with MariaDB 10.4.7: "Incorrect index name 'ACCESS_TOKEN_ID'"

### ISSUE THREAD

**Issue by ytakahashi1981** [gitbucket/gitbucket#2361]
## Issue (Japanese below.)

**Impacted version**: 4.32.0
**MariaDB Version**: 10.4.7
**Deployment mode**: with Jetty 9.4.8

**Problem description**:
I tried to migrate database from h2database to mysql(mariadb), followed https://github.com/gitbucket/gitbucket/wiki/External-database-configuration (also https://takezoe.hatenablog.com/entry/2016/07/16/025154).
But after rebooting jetty, the following stacktrace has shown and GitBucket doesn't work.

```
. 2019-08-15 11:51:59.217:WARN:oejw.WebAppContext:main: Failed startup of context o.e.j.w.WebAppContext@3159c4b8{/gitbucket,file:///tmp/jetty-0.0.0.0-8080-gitbucket.war-_gitbucket-any-198649221540653045.dir/webapp/,UNAVAILABLE}{/gitbucket.war}
java.sql.SQLSyntaxErrorException: (conn=54140) Incorrect index name 'ACCESS_TOKEN_ID'
	at org.mariadb.jdbc.internal.util.exceptions.ExceptionMapper.get(ExceptionMapper.java:242)
	at org.mariadb.jdbc.internal.util.exceptions.ExceptionMapper.getException(ExceptionMapper.java:171)
	at org.mariadb.jdbc.MariaDbStatement.executeExceptionEpilogue(MariaDbStatement.java:248)
	at org.mariadb.jdbc.ClientSidePreparedStatement.executeInternal(ClientSidePreparedStatement.java:230)
	at org.mariadb.jdbc.ClientSidePreparedStatement.execute(ClientSidePreparedStatement.java:157)
	at org.mariadb.jdbc.ClientSidePreparedStatement.executeUpdate(ClientSidePreparedStatement.java:192)
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java)
	at io.github.gitbucket.solidbase.migration.MigrationUtils.updateDatabase(MigrationUtils.java:20)
	at io.github.gitbucket.solidbase.migration.LiquibaseMigration.migrate(LiquibaseMigration.java:93)
	at io.github.gitbucket.solidbase.migration.LiquibaseMigration.migrate(LiquibaseMigration.java:54)
	at io.github.gitbucket.solidbase.Solidbase.migrate(Solidbase.java:45)
	at io.github.gitbucket.solidbase.Solidbase.migrate(Solidbase.java:29)
	at gitbucket.core.servlet.InitializeListener.$anonfun$contextInitialized$1(InitializeListener.scala:73)
	at gitbucket.core.servlet.InitializeListener.$anonfun$contextInitialized$1$adapted(InitializeListener.scala:63)
	at com.github.takezoe.slick.blocking.BlockingJdbcProfile$BlockingAPI$BlockingDatabase.$anonfun$withTransaction$2(BlockingProfile.scala:207)
	at slick.JdbcProfileBlockingSession$BlockingSession.withTransaction(TransactionalJdbcBackend.scala:26)
	at com.github.takezoe.slick.blocking.BlockingJdbcProfile$BlockingAPI$BlockingDatabase.$anonfun$withTransaction$1(BlockingProfile.scala:207)
	at com.github.takezoe.slick.blocking.BlockingJdbcProfile$BlockingAPI$BlockingDatabase.withSession(BlockingProfile.scala:200)
	at com.github.takezoe.slick.blocking.BlockingJdbcProfile$BlockingAPI$BlockingDatabase.withTransaction(BlockingProfile.scala:207)
	at gitbucket.core.servlet.InitializeListener.contextInitialized(InitializeListener.scala:63)
```

I tried below, but not worked.(same stacktrace has shown.):
* delete database and retried rebooting jetty

Doesn't GitBucket support MariaDB 1.4.7?

----

H2DBからmysql(mariadb)へ移行しようと
https://github.com/gitbucket/gitbucket/wiki/External-database-configuration
https://takezoe.hatenablog.com/entry/2016/07/16/025154
を参考に作業しましたが、Jetty起動時に例外が表示されてしまい起動できません。
(例外の内容は省略)

一度DBを削除して再実行してみても同様の挙動でした。

MariaDB10.4.7はサポートの対象外でしょうか？

**Comment by satodoc** [comment]
I have the same problem.

When I examined every version, it did not occur in mariadb: 10.4.3, but it seems to occur from mariadb: 10.4.4 or later.

An error occurred initiaze prorcess in the following.
```xml:
  <createTable tableName="ACCESS_TOKEN">
    <column name="ACCESS_TOKEN_ID" type="int" nullable="false" autoIncrement="true" unique="true"/>
    <column name="TOKEN_HASH" type="varchar(40)" nullable="false"/>
    <column name="USER_NAME" type="varchar(100)" nullable="false"/>
    <column name="NOTE" type="text" nullable="false"/>
  </createTable>


  <addPrimaryKey constraintName="IDX_ACCESS_TOKEN_PK" tableName="ACCESS_TOKEN" columnNames="ACCESS_TOKEN_ID"/>
```

```SQL
ALTER TABLE gitbucket.ACCESS_TOKEN ADD PRIMARY KEY (ACCESS_TOKEN_ID);
```

**Comment by codeling** [comment]
Running into the same problem here... does anybody know, is this a mariadb issue? Does anybody know a fix/workaround?

**Comment by codeling** [comment]
Any news on this?

It's probably out of the scope of this issue, but are there any long term plans regarding supporting MariaDB "officially"? To me MariaDB seems to be a bit more open than MySQL (in terms of for example how they handle their issue tracking and who can contribute in development).

In the meantime, I added a note on this to the migration guide (https://github.com/gitbucket/gitbucket/wiki/External-database-configuration).

**Comment by skullwritter** [comment]
@codeling, @satodoc replyed with the solution:
`ALTER TABLE gitbucket.ACCESS_TOKEN ADD PRIMARY KEY (ACCESS_TOKEN_ID);`

**Comment by takezoe** [comment]
Fixed in #2435
